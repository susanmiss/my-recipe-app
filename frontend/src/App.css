body{
  font-family: 'Nothing You Could Do', cursive;
  font-size: 30px;
  background-color: rgb(70, 68, 68);
  color: white;
}

hr{
  color: white;
}

h1{
  text-align: center;
  font-size: 60px;
}

h2{
  text-align: center;
  margin-top: 40px;
}


li{
  padding-left: 50px;
  color: white;
}

a{
  color: white;
}

a:hover{
  color: rgb(63, 14, 14)
}

.logo-img{
  border-radius: 15px;
  max-width: 50%;
  display: block;
  margin-left: auto;
  margin-right: auto;
}

