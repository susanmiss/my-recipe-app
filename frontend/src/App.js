import React from 'react';
import {BrowserRouter, Switch, Route} from 'react-router-dom';
import './App.css';
import Home from './containers/Home'
import CreateRecipe from './components/CreateRecipe';
import SingleRecipe from './components/SingleRecipe';
import UpdateRecipe from './components/UpdateRecipe';
import Nav from './components/Nav'

function App() {
  return (
    <BrowserRouter>
    <Nav />
        <Switch> 
            <Route path='/' exact component={Home} />
            <Route path='/create' exact component={CreateRecipe} />
            <Route path='/recipe/:id' exact component={SingleRecipe} />
            <Route path='/recipe/update/:id' exact component={UpdateRecipe} />
        </Switch>   
    </BrowserRouter>
  );
}

export default App;
