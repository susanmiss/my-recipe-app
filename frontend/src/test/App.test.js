import React from 'react';
import ReactDOM from 'react-dom'
import EnzymeAdapter from 'enzyme-adapter-react-16'
import {shallow} from 'enzyme'; 

import renderer from 'react-test-renderer';
import App from '../App'


const wrapper = shallow(<App />)
  
test('mounts to DOM', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
});

test('renders correctly', () => {
    const tree = renderer
        .create(<App />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

test('renders widhout crashing', () => {
  console.log(wrapper.debug()); 
  expect(wrapper).toBeTruthy(); //see if teh wrapper exists.
});

test('should render a BrowserRouter component', () =>{
    expect(wrapper.find('BrowserRouter').length).toEqual(1);
})

test('should render a Switch component', () =>{
    expect(wrapper.find('Switch').length).toEqual(1);
})

test('renders <Nav/>', () => {
  const navComponent = wrapper.find("Nav")
  expect(navComponent.length).toBe(1);
})


test('renders <Route/>', () => {
    const routeComponent = wrapper.find("Route")
    expect(routeComponent.length).toBe(4);
})