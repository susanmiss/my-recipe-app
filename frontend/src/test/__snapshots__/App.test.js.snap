// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <nav
    className="nav"
  >
    <ul
      className="nav nav-tabs"
    >
      <li
        className="nav-item pr-3 pt-3 pb-3"
      >
        <a
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li
        className="nav-item pr-3 pt-3 pb-3"
      >
        <a
          href="/create"
          onClick={[Function]}
        >
          Create
        </a>
      </li>
    </ul>
  </nav>,
  <div
    className="container pb-5"
  >
    <br />
    <br />
    <h1
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      Getting fat in lockdown
    </h1>
    <h2>
      Share only FAT recipes that can make us very fat :)
    </h2>
    <br />
    <br />
    <img
      alt="logo"
      className="logo-img"
      src="logo.jpg"
    />
    <hr />
  </div>,
]
`;
